security-analysis/security-analysis-api/src/main/java/com/powsybl/security/action/GeneratorAction.java

```diff
@@ -0,0 +1,91 @@
+/**
+ * Copyright (c) 2022, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.powsybl.security.action;
+
+import java.util.Objects;
+import java.util.Optional;
+
+/**
+ * An action to:
+ * - increase or decrease the targetV of a generator.
+ * - move to voltage control or remove from voltage control.
+ * - change targetQ of a generator that is not a voltage controller.
+ * - change targetV of a generator that is a voltage controller.
+ *
+ * @author Hadrien Godard <hadrien.godard@artelys.com>
+ * @author Anne Tilloy <anne.tilloy@rte-france.com>
+ */
+public class GeneratorAction extends AbstractAction {
+
+    public static final String NAME = "GENERATOR";
+
+    private final String generatorId;
+    private Boolean activePowerRelativeValue; // true if it is a relative variation, false if it is a new targetP
+    private Double activePowerValue; // could be a new targetP if relativeVariation equals false or a relative variation of targetP.
+    private Boolean voltageRegulatorOn;
+    private Double targetV; // absolute value only.
+    private Double targetQ; // absolute value only.
+
+    public GeneratorAction(String id, String generatorId) {
+        super(id);
+        this.generatorId = Objects.requireNonNull(generatorId);
+    }
+
+    @Override
+    public String getType() {
+        return NAME;
+    }
+
+    public String getGeneratorId() {
+        return generatorId;
+    }
+
+    public Optional<Boolean> isActivePowerRelativeValue() {
+        return Optional.ofNullable(activePowerRelativeValue);
+    }
+
+    public GeneratorAction setActivePowerRelativeValue(Boolean activePowerRelativeValue) {
+        this.activePowerRelativeValue = activePowerRelativeValue;
+        return this;
+    }
+
+    public Optional<Double> getActivePowerValue() {
+        return Optional.ofNullable(activePowerValue);
+    }
+
+    public GeneratorAction setActivePowerValue(Double activePowerValue) {
+        this.activePowerValue = activePowerValue;
+        return this;
+    }
```

Comment:
I see a risk of API users setting activePowerValue but forgetting to set if it is relative or absolute.

Maybe we should provide a single api as follows to force users specifying both values, using primitive types (non nullable), and delete above setActivePowerRelativeValue method.

```suggestion
    public GeneratorAction setActivePowerValue(double activePowerValue, boolean relative) {
        this.activePowerValue = activePowerValue;
        this.activePowerRelative= relative;
        return this;
    }
    
    public GeneratorAction unsetActivePowerValue() {
        this.activePowerValue = null;
        this.activePowerRelative= null;
        return this;
    }
```

Reply:
Great idea.

Reply:
Optionals are generally used for return types (to allow method chaining), not input types: it's considered better to have primitive or boxed types as inputs/class fields.